File: cpp/266. Palindrome Permutation.cpp
class Solution {
public:
    bool canPermutePalindrome(string s) {
        unordered_map<char, int> map;
        
        for(int i = 0; i < s.length(); ++i){
            if(map[s[i]] == 0) map[s[i]]++;
            else map[s[i]]--;
        }
        int count = 0;
        for(int i = 0; i < 256; ++i){
            if(map[i] == 1) count ++;
        }
        
        return count <=1;
    }
};

File: java/266. Palindrome Permutation.java
public class Solution {
    public boolean canPermutePalindrome(String s) {
        Map<Character, Integer> map = new HashMap<Character, Integer> ();
        int len = s.length();
        for(int i = 0; i < len; ++i){
            char c = s.charAt(i);
            Integer count = map.get(c);
            if(count == null){
                count = new Integer(0);
            }
            map.put(c,++count);
        }
        
        boolean hasOdd = false;
        for(Character c : map.keySet()){
            if(map.get(c) % 2 == 1){
                if(!hasOdd){
                    hasOdd = true;
                }else{
                    return false;
                }
            }
        }
        return true;
    }
}